index.bs

```diff
@@ -432,10 +434,29 @@ the computation graph used to compute one or more specified outputs. The key
 purpose of the compilation step is to enable optimizations that span two or
 more operations, such as operation or loop fusion.
 
-The {{MLGraph/compute()}} method of the {{MLGraph}} interface is used to execute the
-compiled computation graph (to perform inference). The caller supplies the input
-values using {{MLNamedInputs}}, binding the input {{MLOperand}}s to their values.
-The caller supplies pre-allocated buffers for output {{MLOperand}}s using {{MLNamedOutputs}}.
+Once the {{MLGraph}} is constructed, there are multiple ways by which the graph may be executed.
+The {{MLExecution}} interface represents a way the execution of the graph is carried out immediately on the
+calling thread, which must also be a worker thread. The execution is carried out by the {{MLExecution/compute()}},
+method which produces the results of the computation from all the inputs bound to graph on the bound outputs.
+This type of execution is limited to only when the computational device bound to the ML context is a CPU.
```

Comment:
That was my original thought, but @RafaelCintron believes that we should not allow sync execution on a GPU context, because it means you need to block on the calling thread until the GPU executes and produces the results in the output tensors, which could create jarring user experience.

Reply:
if this is restricted to  `DedicatedWorker`, I'm not sure I understand where the risk with blocking on a GPU execution would be. Having it applicable to GPU contexts would help with [the concern I raised](https://github.com/webmachinelearning/webnn/pull/255#issuecomment-1057762561)

Reply:
> which could create jarring user experience.

Because it is limited in the worker, I suppose the UI thread (main thread) won't be blocked in this way.